internal/sops/azkv/keysource.go

```diff
@@ -177,3 +189,82 @@ func decode(b []byte) ([]byte, error) {
 	}
 	return ioutil.ReadAll(reader)
 }
+
+// getTokenCredential returns the tokenCredential of the MasterKey, or
+// azidentity.NewDefaultAzureCredential.
+func (key *MasterKey) getTokenCredential() (azcore.TokenCredential, error) {
+	if key.token == nil {
+		return getDefaultAzureCredential()
+	}
+	return key.token, nil
+}
+
+// Function copied from parts of `NewDefaultAzureCredential` in
```

Comment:
This gives me a "where it was copied from", but not a "why it was partly copied" answer.